pkg/binding/binding-controller.go

```diff
@@ -601,6 +602,23 @@ func (c *Controller) setupBindingInformer(ctx context.Context) error {
 	return nil
 }
 
+func shouldQueueBindingOnUpdate(old, new *v1alpha1.Binding) bool {
+	if old.Generation != new.Generation {
+		return true
+	}
+	if len(old.Status.Conditions) != len(new.Status.Conditions) || len(old.Status.Errors) != len(new.Status.Errors) {
```

Comment:
FYI, `abstract.SliceEqual` checks for equal lengths, among other things.

Reply:
Improved.

BTW glad to see your recent contribution to `abstract.SliceEqual`!